src/Stuart/Client.php

```diff
@@ -11,9 +11,9 @@ class Client
     private $jobRepository;
     private $jobPricingRepository;
 
-    public function __construct($authenticator)
+    public function __construct($authenticator, \GuzzleHttp\Client $client = null)
```

Comment:
Would it be possible to add a new `options` parameter to the constructor in order to allow customization? Example:
```php
 public function __construct($authenticator, $options = array())
{
  $logApiCalls = $options['log_api_calls'];
  ...
}
```

Reply:
I think your suggestion is too restrictive in terms of customization, as it would impose users a specific logging format. 

While I don't really mind that the library provides some built-in logging, it most likely won't fit my requirements, and I think the simplest/most flexible solution in terms of maintenance for everyone is to expose the underlying Guzzle client either via constructor injection, or via a getter so it becomes possibles to bind middlewares.

Reply:
Yes you are right, it just feels weird now to have the `GuzzleHttp\Client` given to the `Client` and `HttpClient` constructors 🤔

But as we want the `HttpClient` to be used independently from the `Client` (to use endpoints that are not covered yet by the `Client` for example) 👍